lms/djangoapps/grades/new/subsection_grade.py

```diff
@@ -10,13 +13,28 @@
 from lms.djangoapps.grades.config.models import PersistentGradesEnabledFlag
 from student.models import anonymous_id_for_user, User
 from submissions import api as submissions_api
+from traceback import format_exc
 from xmodule import block_metadata_utils, graders
 from xmodule.graders import Score
 
 
 log = getLogger(__name__)
 
 
+@contextmanager
+def persistence_safe_fallback():
+    """
+    A context manager intended to be used to wrap robust grades database-specific code that can be ignored on failure.
+    """
+    try:
+        with transaction.atomic():
+            yield
+    except IntegrityError:
```

Comment:
![image](https://cloud.githubusercontent.com/assets/7373924/18489051/f736be60-79c9-11e6-8317-59c70b824f77.png)

We discussed this, then I immediately went and did the wrong thing. Will fix.